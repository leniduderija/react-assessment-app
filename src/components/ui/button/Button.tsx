import React, { ButtonHTMLAttributes, MouseEvent } from "react";
import "./Button.css";
import cn from "classnames";
import { StyledButton } from "./Button.styled";

export enum ButtonShape {
  Rounded = "rounded",
  Square = "square",
}

export enum ButtonVariant {
  Solid = "solid",
  Outlined = "outlined",
  Link = "link",
}

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  variant?: ButtonVariant;
  shape?: ButtonShape;
  color?: string;
}

export const Button = ({
  onClick,
  variant = ButtonVariant.Solid,
  shape = ButtonShape.Square,
  color,
  children,
  className,
  ...props
}: ButtonProps) => {
  return (
    <StyledButton
      className={cn("Button", className)}
      onClick={onClick}
      {...props}
      data-testid="Button"
      shape={shape}
      variant={variant}
      color={color}
    >
      {children}
    </StyledButton>
  );
};
