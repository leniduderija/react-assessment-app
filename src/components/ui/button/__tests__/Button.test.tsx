import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Button component', () => {
  test('renders correctly', () => {
    render(<Button>Click me</Button>);
    const buttonElement = screen.getByTestId('Button');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent('Click me');
  });

  test('applies className correctly', () => {
    render(<Button className="custom-class">Click me</Button>);
    const buttonElement = screen.getByTestId('Button');
    expect(buttonElement).toHaveClass('custom-class');
  });

  test('fires onClick event', () => {
    const onClickMock = jest.fn();
    render(<Button onClick={onClickMock}>Click me</Button>);
    const buttonElement = screen.getByTestId('Button');
    buttonElement.click();
    expect(onClickMock).toHaveBeenCalled();
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<Button>Click me</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
