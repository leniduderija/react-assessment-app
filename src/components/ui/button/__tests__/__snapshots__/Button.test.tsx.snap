// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<button
  className="sc-beySPh cnvGWG Button"
  data-testid="Button"
  shape="square"
  variant="solid"
>
  Click me
</button>
`;
